tests/js/client/server_parameters/test-export-shard-usage-metrics-bytes-per-shard-per-user-cluster.js

```diff
@@ -0,0 +1,1837 @@
+/*jshint globalstrict:false, strict:false */
+/* global getOptions, assertEqual, assertNotEqual, assertTrue, assertFalse, arango */
+
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2024, ArangoDB Inc, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB Inc, Cologne, Germany
+///
+/// @author Jan Steemann
+/// @author Copyright 2024, ArangoDB Inc, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+const jwtSecret = 'abc123';
+
+if (getOptions === true) {
+  return {
+    'server.export-shard-usage-metrics': "enabled-per-shard-per-user",
+    'server.authentication': 'true',
+    'server.jwt-secret': jwtSecret,
+  };
+}
+
+const jsunity = require('jsunity');
+const db = require('@arangodb').db;
+const { getDBServers, deriveTestSuite } = require("@arangodb/test-helper");
+const request = require("@arangodb/request");
+const users = require("@arangodb/users");
+const crypto = require('@arangodb/crypto');
+
+const jwt = crypto.jwtEncode(jwtSecret, {
+  "server_id": "ABCD",
+  "iss": "arangodb", "exp": Math.floor(Date.now() / 1000) + 3600
+}, 'HS256');
+
+function BaseTestSuite(targetUser) {
+  const baseName = "UnitTestsCollection";
+  let nextCollectionId = 0;
+
+  let getUniqueCollectionName = () => {
+    return baseName + nextCollectionId++;
+  };
+
+  let getRawMetrics = function() {
+    let lines = [];
+    getDBServers().forEach((server) => {
+      let res = request({ method: "GET", url: server.url + "/_admin/usage-metrics", auth: { bearer: jwt } });
+      assertEqual(200, res.status);
+      lines = lines.concat(res.body.split(/\n/).filter((l) => l.match(/^arangodb_collection_requests_bytes_(read|written)_total/)));
+    });
+    return lines;
+  };
+
+  let getParsedMetrics = function(database, collection) {
```

Comment:
Helper methods in seperate module, like mentioned before in file: `tests/js/client/server_parameters/test-export-shard-usage-metrics-bytes-per-shard-cluster.js` would help maintainability/readability. (but "minor").